Installation issue: hipsycl (incompatibility with llvm 20)

### Steps to reproduce the issue

```console
spack spec -I hipsycl
 -   hipsycl@24.06.0~cuda~ipo~rocm build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]      ^boost@1.69.0~atomic~chrono~clanglibcpp~container+context~contract~coroutine~date_time~debug~exception+fiber+filesystem~graph~graph_parallel~icu~iostreams~json~locale~log~math~mpi+multithreaded~nowide~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded~stacktrace~system~taggedlayout~test~thread~timer~type_erasure~url~versionedlayout~wave build_system=generic context-impl=fcontext cxxstd=17 patches=3a83d90,5da7ad2,607b077 visibility=hidden arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]      ^cmake@3.31.6~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^curl@8.11.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^nghttp2@1.65.0 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^zlib-ng@2.2.3+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]      ^gcc-runtime@9.4.0 build_system=generic arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[e]      ^glibc@2.31 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]      ^gmake@4.4.1~guile build_system=generic arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]      ^llvm@20.1.0+clang~cuda~flang+gold~ipo+libomptarget~libomptarget_debug~link_llvm_dylib+lld+lldb+llvm_dylib+lua~mlir+offload+polly~python~split_dwarf~z3~zstd build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=runtime openmp=runtime shlib_symbol_version=none targets=all version_suffix=none arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^binutils@2.43.1~debuginfod~gas+gold~gprofng+headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^diffutils@3.10 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^zstd@1.5.6+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^hwloc@2.11.1~cairo~cuda~gl~level_zero~libudev+libxml2~nvml~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^libpciaccess@0.17 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]                  ^util-macros@1.20.1 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^libedit@3.1-20240808 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^libffi@3.4.6 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^libxml2@2.13.5~http+pic~python+shared build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^libiconv@1.17 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^lua@5.3.6+shared build_system=makefile fetcher=curl arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^unzip@6.0 build_system=makefile patches=881d2ed,f6f6236 arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^ninja@1.12.1+re2c build_system=generic patches=93f4bb3 arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^re2c@3.1 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^perl-data-dumper@2.173 build_system=perl arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]                  ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^pkgconf@2.3.0 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^swig@4.1.1 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^pcre2@10.44~jit+multibyte+pic build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^xz@5.6.3~pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]      ^python@3.13.2+bz2+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^expat@2.6.4+libbsd build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^libbsd@0.12.2 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]                  ^libmd@1.1.0 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^gdbm@1.23 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^tar@1.35 build_system=autotools zip=pigz arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]                  ^pigz@2.8 build_system=makefile arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^openssl@3.4.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]              ^ca-certificates-mozilla@2025-02-25 build_system=generic arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^readline@8.2 build_system=autotools patches=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^sqlite@3.46.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
[+]          ^util-linux-uuid@2.40.2 build_system=autotools arch=linux-ubuntu20.04-skylake %gcc@9.4.0
```


### Error message

<details><summary>Error message</summary>

<pre>
51 errors found in build log:
     293     [ 33%] Built target libkernel-sscp-host-fast
     294     [ 35%] Building CXX object src/compiler/CMakeFiles/acpp-clang-cbs.dir/cbs/LoopsParallelMarker.cpp.o
     295     [ 35%] Building CXX object src/compiler/CMakeFiles/acpp-clang-cbs.dir/cbs/PHIsToAllocas.cpp.o
     296     cd /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-build-jicrqsg/src/compiler && /opt/spack/lib/spack/e
             nv/gcc/g++ -DHIPSYCL_COMPILER_COMPONENT -DHIPSYCL_DEBUG_LEVEL=2 -DHIPSYCL_WITH_ACCELERATED_CPU -DHIPSYCL_WITH_REFLECTION_BUILTINS -DHIPSYCL_WITH
             _SSCP_COMPILER -DHIPSYCL_WITH_STDPAR_COMPILER -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/root/sp
             ack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include -I/tmp/root/spack-stage/spack-stage-
             hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler -I/tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzf
             enp3jpx7hirz/spack-build-jicrqsg/include -isystem /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752t
             cs7sfbv/include -O3 -DNDEBUG -std=c++17 -fPIC -MD -MT src/compiler/CMakeFiles/acpp-clang-cbs.dir/cbs/PHIsToAllocas.cpp.o -MF CMakeFiles/acpp-cla
             ng-cbs.dir/cbs/PHIsToAllocas.cpp.o.d -o CMakeFiles/acpp-clang-cbs.dir/cbs/PHIsToAllocas.cpp.o -c /tmp/root/spack-stage/spack-stage-hipsycl-24.06
             .0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/PHIsToAllocas.cpp
     297     cd /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-build-jicrqsg/src/compiler && /opt/spack/lib/spack/e
             nv/gcc/g++ -DHIPSYCL_COMPILER_COMPONENT -DHIPSYCL_DEBUG_LEVEL=2 -DHIPSYCL_WITH_ACCELERATED_CPU -DHIPSYCL_WITH_REFLECTION_BUILTINS -DHIPSYCL_WITH
             _SSCP_COMPILER -DHIPSYCL_WITH_STDPAR_COMPILER -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/root/sp
             ack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include -I/tmp/root/spack-stage/spack-stage-
             hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler -I/tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzf
             enp3jpx7hirz/spack-build-jicrqsg/include -isystem /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752t
             cs7sfbv/include -O3 -DNDEBUG -std=c++17 -fPIC -MD -MT src/compiler/CMakeFiles/acpp-clang-cbs.dir/cbs/LoopsParallelMarker.cpp.o -MF CMakeFiles/ac
             pp-clang-cbs.dir/cbs/LoopsParallelMarker.cpp.o.d -o CMakeFiles/acpp-clang-cbs.dir/cbs/LoopsParallelMarker.cpp.o -c /tmp/root/spack-stage/spack-s
             tage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/LoopsParallelMarker.cpp
     298     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalInliningAttribut
             orPass.cpp: In member function 'llvm::PreservedAnalyses hipsycl::compiler::GlobalInliningAttributorPass::run(llvm::Module&, llvm::ModuleAnalysis
             Manager&)':
  >> 299     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalInliningAttribut
             orPass.cpp:25:18: error: invalid use of incomplete type 'class llvm::Module'
     300        25 |     if(auto* F = M.getFunction(KN))
     301           |                  ^
     302     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     303                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalInliningAttributorPass.hpp:14,
     304                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalInliningAttributorPass.cpp:11:
     305     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     306        20 | class Module;
     307           |       ^~~~~~
  >> 308     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalInliningAttribut
             orPass.cpp:28:17: error: range-based 'for' expression of type 'llvm::Module' has incomplete type
     309        28 |   for(auto &F : M) {
     310           |                 ^
  >> 311     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalInliningAttribut
             orPass.cpp:33:28: error: 'llvm::GlobalValue' has not been declared
     312        33 |         F.setLinkage(llvm::GlobalValue::InternalLinkage);
     313           |                            ^~~~~~~~~~~
  >> 314     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalInliningAttribut
             orPass.cpp:34:36: error: 'llvm::Attribute' has not been declared
     315        34 |         if(!F.hasFnAttribute(llvm::Attribute::AlwaysInline))
     316           |                                    ^~~~~~~~~
  >> 317     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalInliningAttribut
             orPass.cpp:35:29: error: 'llvm::Attribute' has not been declared
     318        35 |           F.addFnAttr(llvm::Attribute::AlwaysInline);
     319           |                             ^~~~~~~~~
  >> 320     make[2]: *** [src/compiler/llvm-to-backend/CMakeFiles/llvm-to-backend.dir/build.make:138: src/compiler/llvm-to-backend/CMakeFiles/llvm-to-backen
             d.dir/GlobalInliningAttributorPass.cpp.o] Error 1
     321     make[2]: *** Waiting for unfinished jobs....
     322     [ 36%] Building CXX object src/compiler/CMakeFiles/acpp-clang-cbs.dir/cbs/RemoveBarrierCalls.cpp.o
     323     cd /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-build-jicrqsg/src/compiler && /opt/spack/lib/spack/e
             nv/gcc/g++ -DHIPSYCL_COMPILER_COMPONENT -DHIPSYCL_DEBUG_LEVEL=2 -DHIPSYCL_WITH_ACCELERATED_CPU -DHIPSYCL_WITH_REFLECTION_BUILTINS -DHIPSYCL_WITH
             _SSCP_COMPILER -DHIPSYCL_WITH_STDPAR_COMPILER -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/root/sp
             ack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include -I/tmp/root/spack-stage/spack-stage-
             hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler -I/tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzf
             enp3jpx7hirz/spack-build-jicrqsg/include -isystem /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752t
             cs7sfbv/include -O3 -DNDEBUG -std=c++17 -fPIC -MD -MT src/compiler/CMakeFiles/acpp-clang-cbs.dir/cbs/RemoveBarrierCalls.cpp.o -MF CMakeFiles/acp
             p-clang-cbs.dir/cbs/RemoveBarrierCalls.cpp.o.d -o CMakeFiles/acpp-clang-cbs.dir/cbs/RemoveBarrierCalls.cpp.o -c /tmp/root/spack-stage/spack-stag
             e-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/RemoveBarrierCalls.cpp
     324     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp: In function 'bool hipsycl::compiler::insertRangeAssumptionForBuiltinCalls(llvm::Module&, llvm::StringRef, long long int, long long int
             , bool)':
     325     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:24:54: warning: 'llvm::Function* llvm::Intrinsic::getDeclaration(llvm::Module*, llvm::Intrinsic::ID, llvm::ArrayRef<llvm::Type*>)' is d
             eprecated: Use getOrInsertDeclaration instead [-Wdeprecated-declarations]
     326        24 |   llvm::Function *AssumeIntrinsic = llvm::Intrinsic::getDeclaration(&M, llvm::Intrinsic::assume);
     ...
     343           |                                                                                                ^
     344     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Inst
             ructions.h:33,
     345                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalSizesFitInI32OptPass.cpp:13:
     346     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Intrinsics.h:102:20: note:
              declared here
     347       102 |   inline Function *getDeclaration(Module *M, ID id, ArrayRef<Type *> Tys = {}) {
     348           |                    ^~~~~~~~~~~~~~
  >> 349     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:28:16: error: invalid use of incomplete type 'class llvm::Module'
     350        28 |   if(auto* F = M.getFunction(BuiltinName)) {
     351           |                ^
     352     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     353                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     354                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalSizesFitInI32OptPass.cpp:11:
     355     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     356        20 | class Module;
     357           |       ^~~~~~
  >> 358     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:40:36: error: invalid use of incomplete type 'class llvm::Module'
     359        40 |             llvm::ConstantInt::get(M.getContext(),
     360           |                                    ^
     361     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     362                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     363                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalSizesFitInI32OptPass.cpp:11:
     364     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     365        20 | class Module;
     366           |       ^~~~~~
  >> 367     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:49:36: error: invalid use of incomplete type 'class llvm::Module'
     368        49 |             llvm::ConstantInt::get(M.getContext(),
     369           |                                    ^
     370     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     371                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     372                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalSizesFitInI32OptPass.cpp:11:
     373     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     374        20 | class Module;
     375           |       ^~~~~~
  >> 376     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:54:79: error: no matching function for call to 'llvm::SmallVector<llvm::Value*>::SmallVector(<brace-enclosed initializer list>)'
     377        54 |         llvm::SmallVector<llvm::Value*> CallArgGreaterEqualMin{GreaterEqualMin};
     378           |                                                                               ^
     379     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/Arr
             ayRef.h:13,
     380                      from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/Tin
             yPtrVector.h:12,
     381                      from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:43,
     382                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     ...
     418           |            ^~~~~~~~~~~
     419     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/SmallVector.h:1205:12: no
             te:   conversion of argument 1 would be ill-formed:
     420     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/SmallVector.h:1198:3: not
             e: candidate: 'llvm::SmallVector<T, N>::SmallVector() [with T = llvm::Value*; unsigned int N = 6]'
     421      1198 |   SmallVector() : SmallVectorImpl<T>(N) {}
     422           |   ^~~~~~~~~~~
     423     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/SmallVector.h:1198:3: not
             e:   candidate expects 0 arguments, 1 provided
  >> 424     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:55:75: error: no matching function for call to 'llvm::SmallVector<llvm::Value*>::SmallVector(<brace-enclosed initializer list>)'
     425        55 |         llvm::SmallVector<llvm::Value*> CallArgLesserThanMax{LesserThanMax};
     426           |                                                                           ^
     427     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/Arr
             ayRef.h:13,
     428                      from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/Tin
             yPtrVector.h:12,
     429                      from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:43,
     430                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     ...
     467     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/SmallVector.h:1205:12: no
             te:   conversion of argument 1 would be ill-formed:
     468     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/SmallVector.h:1198:3: not
             e: candidate: 'llvm::SmallVector<T, N>::SmallVector() [with T = llvm::Value*; unsigned int N = 6]'
     469      1198 |   SmallVector() : SmallVectorImpl<T>(N) {}
     470           |   ^~~~~~~~~~~
     471     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/ADT/SmallVector.h:1198:3: not
             e:   candidate expects 0 arguments, 1 provided
     472     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp: In member function 'llvm::PreservedAnalyses hipsycl::compiler::GlobalSizesFitInI32OptPass::run(llvm::Module&, llvm::ModuleAnalysisMana
             ger&)':
  >> 473     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:75:16: error: invalid use of incomplete type 'class llvm::Module'
     474        75 |   if(auto* F = M.getFunction(IfFitsInIntBuiltinName)) {
     475           |                ^
     476     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     477                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     478                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalSizesFitInI32OptPass.cpp:11:
     479     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     480        20 | class Module;
     481           |       ^~~~~~
  >> 482     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:79:36: error: invalid use of incomplete type 'class llvm::Module'
     483        79 |           llvm::BasicBlock::Create(M.getContext(), "", F);
     484           |                                    ^
     485     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     486                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     487                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalSizesFitInI32OptPass.cpp:11:
     488     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     489        20 | class Module;
     490           |       ^~~~~~
  >> 491     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:81:11: error: invalid use of incomplete type 'class llvm::Module'
     492        81 |           M.getContext(),
     493           |           ^
     494     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     495                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     496                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalSizesFitInI32OptPass.cpp:11:
     497     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     498        20 | class Module;
     499           |       ^~~~~~
  >> 500     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/GlobalSizesFitInI32Opt
             Pass.cpp:83:15: error: invalid use of incomplete type 'class llvm::Module'
     501        83 |               M.getContext(),
     502           |               ^
     503     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     504                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/GlobalSizesFitInI32OptPass.hpp:14,
     505                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             GlobalSizesFitInI32OptPass.cpp:11:
     506     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     507        20 | class Module;
     508           |       ^~~~~~
  >> 509     make[2]: *** [src/compiler/llvm-to-backend/CMakeFiles/llvm-to-backend.dir/build.make:124: src/compiler/llvm-to-backend/CMakeFiles/llvm-to-backen
             d.dir/GlobalSizesFitInI32OptPass.cpp.o] Error 1
     510     In file included from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnno
             tationAnalysis.cpp:13:
     511     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hipSYCL/compiler/cbs/IRU
             tils.hpp: In function 'void hipsycl::compiler::utils::findFunctionsWithStringAnnotationsWithArg(llvm::Module&, Handler&&)':
     512     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hipSYCL/compiler/cbs/IRU
             tils.hpp:194:18: warning: invalid use of incomplete type 'class llvm::Module'
     513       194 |   for (auto &I : M.globals()) {
     514           |                  ^
     515     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Lega
             cyPassManagers.h:20,
     516                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hi
             pSYCL/compiler/cbs/SplitterAnnotationAnalysis.hpp:14,
     517                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnno
             tationAnalysis.cpp:11:
     518     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/Pass.h:42:7: note: forward de
             claration of 'class llvm::Module'
     519        42 | class Module;
     520           |       ^~~~~~
     521     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnnotationAnalysis.cpp: In
              member function 'bool hipsycl::compiler::SplitterAnnotationInfo::analyzeModule(llvm::Module&)':
  >> 522     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnnotationAnalysis.cpp:27:
             29: error: invalid use of incomplete type 'class llvm::Module'
     523        27 |   if (auto *BarrIntrinsic = M.getFunction(hipsycl::compiler::cbs::BarrierIntrinsicName)) {
     524           |                             ^
     525     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Lega
             cyPassManagers.h:20,
     526                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hi
             pSYCL/compiler/cbs/SplitterAnnotationAnalysis.hpp:14,
     527                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnno
             tationAnalysis.cpp:11:
     528     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/Pass.h:42:7: note: forward de
             claration of 'class llvm::Module'
     529        42 | class Module;
     530           |       ^~~~~~
     531     [ 37%] Building CXX object src/compiler/CMakeFiles/acpp-clang-cbs.dir/cbs/CanonicalizeBarriers.cpp.o
  >> 532     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnnotationAnalysis.cpp:42:
             17: error: invalid use of incomplete type 'class llvm::Module'
     533        42 |   if (auto MD = M.getNamedMetadata(SscpAnnotationsName))
     534           |                 ^
     535     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Lega
             cyPassManagers.h:20,
     536                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hi
             pSYCL/compiler/cbs/SplitterAnnotationAnalysis.hpp:14,
     537                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnno
             tationAnalysis.cpp:11:
     538     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/Pass.h:42:7: note: forward de
             claration of 'class llvm::Module'
     ...
     540           |       ^~~~~~
     541     cd /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-build-jicrqsg/src/compiler && /opt/spack/lib/spack/e
             nv/gcc/g++ -DHIPSYCL_COMPILER_COMPONENT -DHIPSYCL_DEBUG_LEVEL=2 -DHIPSYCL_WITH_ACCELERATED_CPU -DHIPSYCL_WITH_REFLECTION_BUILTINS -DHIPSYCL_WITH
             _SSCP_COMPILER -DHIPSYCL_WITH_STDPAR_COMPILER -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/root/sp
             ack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include -I/tmp/root/spack-stage/spack-stage-
             hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler -I/tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzf
             enp3jpx7hirz/spack-build-jicrqsg/include -isystem /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752t
             cs7sfbv/include -O3 -DNDEBUG -std=c++17 -fPIC -MD -MT src/compiler/CMakeFiles/acpp-clang-cbs.dir/cbs/CanonicalizeBarriers.cpp.o -MF CMakeFiles/a
             cpp-clang-cbs.dir/cbs/CanonicalizeBarriers.cpp.o.d -o CMakeFiles/acpp-clang-cbs.dir/cbs/CanonicalizeBarriers.cpp.o -c /tmp/root/spack-stage/spac
             k-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/CanonicalizeBarriers.cpp
     542     In file included from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnno
             tationAnalysis.cpp:13:
     543     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hipSYCL/compiler/cbs/IRU
             tils.hpp: In instantiation of 'void hipsycl::compiler::utils::findFunctionsWithStringAnnotationsWithArg(llvm::Module&, Handler&&) [with Handler 
             = hipsycl::compiler::utils::findFunctionsWithStringAnnotations(llvm::Module&, Handler&&) [with Handler = hipsycl::compiler::SplitterAnnotationIn
             fo::analyzeModule(llvm::Module&)::<lambda(llvm::Function*, llvm::StringRef)>]::<lambda(llvm::Function*, llvm::StringRef, llvm::Value*)>]':
     544     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hipSYCL/compiler/cbs/IRU
             tils.hpp:214:44:   required from 'void hipsycl::compiler::utils::findFunctionsWithStringAnnotations(llvm::Module&, Handler&&) [with Handler = hi
             psycl::compiler::SplitterAnnotationInfo::analyzeModule(llvm::Module&)::<lambda(llvm::Function*, llvm::StringRef)>]'
     545     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnnotationAnalysis.cpp:40:
             4:   required from here
  >> 546     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hipSYCL/compiler/cbs/IRU
             tils.hpp:194:20: error: invalid use of incomplete type 'class llvm::Module'
     547       194 |   for (auto &I : M.globals()) {
     548           |                  ~~^~~~~~~
     549     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Lega
             cyPassManagers.h:20,
     550                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/../../include/hi
             pSYCL/compiler/cbs/SplitterAnnotationAnalysis.hpp:14,
     551                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/cbs/SplitterAnno
             tationAnalysis.cpp:11:
     552     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/Pass.h:42:7: note: forward de
             claration of 'class llvm::Module'
     553        42 | class Module;
     554           |       ^~~~~~
     555     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/KnownGroupSizeOptPass.
             cpp: In function 'bool hipsycl::compiler::{anonymous}::applyKnownGroupSize(llvm::Module&, int, llvm::StringRef, llvm::StringRef)':
  >> 556     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/KnownGroupSizeOptPass.
             cpp:31:28: error: invalid use of incomplete type 'class llvm::Module'
     557        31 |   if(auto* GetGroupSizeF = M.getFunction(GetGroupSizeBuiltinName)) {
     558           |                            ^
     559     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     560                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/KnownGroupSizeOptPass.hpp:14,
     561                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             KnownGroupSizeOptPass.cpp:11:
     562     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     563        20 | class Module;
     564           |       ^~~~~~
  >> 565     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/KnownGroupSizeOptPass.
             cpp:35:9: error: invalid use of incomplete type 'class llvm::Module'
     566        35 |         M.getOrInsertFunction(NewFunctionName, GetGroupSizeF->getFunctionType(),
     567           |         ^
     568     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     569                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/KnownGroupSizeOptPass.hpp:14,
     570                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             KnownGroupSizeOptPass.cpp:11:
     571     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
     572        20 | class Module;
     573           |       ^~~~~~
  >> 574     /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/KnownGroupSizeOptPass.
             cpp:45:34: error: invalid use of incomplete type 'class llvm::Module'
     575        45 |         llvm::BasicBlock::Create(M.getContext(), "", NewGetGroupSizeF);
     576           |                                  ^
     577     In file included from /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Pass
             Manager.h:44,
     578                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             ../../../include/hipSYCL/compiler/llvm-to-backend/KnownGroupSizeOptPass.hpp:14,
     579                      from /tmp/root/spack-stage/spack-stage-hipsycl-24.06.0-jicrqsgwqbkle43o3vzfenp3jpx7hirz/spack-src/src/compiler/llvm-to-backend/
             KnownGroupSizeOptPass.cpp:11:
     580     /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-20.1.0-2mb5pnljcr2xrddgu6ujl752tcs7sfbv/include/llvm/IR/Analysis.h:20:7: note: for
             ward declaration of 'class llvm::Module'
...
</pre></details>


### Information on your system

```
# spack debug report
* **Spack:** 1.0.0.dev0 (f82de718cd46fe4ad6f179cff605f43dafbbcd1b)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-skylake
```

### Additional information

Adaptivecpp v24.06 is not compatible with llvm 20, but it works with `^llvm@18`.

Also will the hipsycl package get renamed to adaptivecpp ?

[spack-build-out.txt](https://github.com/user-attachments/files/19270811/spack-build-out.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/hipsycl/package.py
+++ b/var/spack/repos/builtin/packages/hipsycl/package.py
@@ -76,6 +76,7 @@ class Hipsycl(CMakePackage, ROCmPackage):
     )
     # https://github.com/spack/spack/issues/46681
     conflicts("^llvm@19", when="@24.02.0:24.06.0")
+    conflicts("^llvm@20", when="@24.02.0:24.06.0")
 
     def cmake_args(self):
         spec = self.spec

--- a/var/spack/repos/builtin/packages/hipsycl/package.py
+++ b/var/spack/repos/builtin/packages/hipsycl/package.py
@@ -24,10 +24,10 @@ class Hipsycl(CMakePackage, ROCmPackage):
     license("BSD-2-Clause")
 
     version("stable", branch="stable", submodules=True)
+    version("24.10.0", commit="7677cf6eefd8ab46d66168cd07ab042109448124", submodules=True)
     version("24.06.0", commit="fc51dae9006d6858fc9c33148cc5f935bb56b075", submodules=True)
     version("24.02.0", commit="974adc33ea5a35dd8b5be68c7a744b37482b8b64", submodules=True)
     version("23.10.0", commit="3952b468c9da89edad9dff953cdcab0a3c3bf78c", submodules=True)
-    version("0.9.4", commit="99d9e24d462b35e815e0e59c1b611936c70464ae", submodules=True)
     version("0.9.4", commit="99d9e24d462b35e815e0e59c1b611936c70464ae", submodules=True)
     version("0.9.3", commit="51507bad524c33afe8b124804091b10fa25618dc", submodules=True)
     version("0.9.2", commit="49fd02499841ae884c61c738610e58c27ab51fdb", submodules=True)
@@ -44,9 +44,28 @@ class Hipsycl(CMakePackage, ROCmPackage):
     depends_on("python@3:")
     depends_on("llvm@8: +clang", when="~cuda")
     depends_on("llvm@9: +clang", when="+cuda")
+
     # hipSYCL 0.8.0 supported only LLVM 8-10:
     # (https://github.com/AdaptiveCpp/AdaptiveCpp/blob/v0.8.0/CMakeLists.txt#L29-L37)
+    # recent versions support only up to llvm18
+    # https://github.com/spack/spack/issues/46681
+    # https://github.com/spack/spack/issues/49506
+    
+    # The following list was made based on the version tested in adaptivecpp github
+    depends_on("llvm@14:18", when="@develop")
+    depends_on("llvm@14:18", when="@stable")
+
+    depends_on("llvm@14:18", when="@24.10.0")
+    depends_on("llvm@14:18", when="@24.06.0")
+    depends_on("llvm@13:17", when="@24.02.0")
+    depends_on("llvm@13:17", when="@23.10.0")
+    depends_on("llvm@11:15", when="@0.9.4")
+    depends_on("llvm@11:14", when="@0.9.3")
+    depends_on("llvm@11:13", when="@0.9.2")
+    depends_on("llvm@11", when="@0.9.1")
+    #depends_on("llvm@10:11", when="@0.9.0") # missing in releases
     depends_on("llvm@8:10", when="@0.8.0")
+
     # https://github.com/spack/spack/issues/45029 and https://github.com/spack/spack/issues/43142
     conflicts("^gcc@12", when="@23.10.0")
     # https://github.com/OpenSYCL/OpenSYCL/pull/918 was introduced after 0.9.4
@@ -74,9 +93,6 @@ class Hipsycl(CMakePackage, ROCmPackage):
         "further info please refer to: "
         "https://github.com/illuhad/hipSYCL/blob/master/doc/install-cuda.md",
     )
-    # https://github.com/spack/spack/issues/46681
-    conflicts("^llvm@19", when="@24.02.0:24.06.0")
-    conflicts("^llvm@20", when="@24.02.0:24.06.0")
 
     def cmake_args(self):
         spec = self.spec

--- a/var/spack/repos/builtin/packages/hipsycl/package.py
+++ b/var/spack/repos/builtin/packages/hipsycl/package.py
@@ -50,7 +50,7 @@ class Hipsycl(CMakePackage, ROCmPackage):
     # recent versions support only up to llvm18
     # https://github.com/spack/spack/issues/46681
     # https://github.com/spack/spack/issues/49506
-    
+
     # The following list was made based on the version tested in adaptivecpp github
     depends_on("llvm@14:18", when="@develop")
     depends_on("llvm@14:18", when="@stable")
@@ -63,7 +63,7 @@ class Hipsycl(CMakePackage, ROCmPackage):
     depends_on("llvm@11:14", when="@0.9.3")
     depends_on("llvm@11:13", when="@0.9.2")
     depends_on("llvm@11", when="@0.9.1")
-    #depends_on("llvm@10:11", when="@0.9.0") # missing in releases
+    # depends_on("llvm@10:11", when="@0.9.0") # missing in releases
     depends_on("llvm@8:10", when="@0.8.0")
 
     # https://github.com/spack/spack/issues/45029 and https://github.com/spack/spack/issues/43142

